pkgs/_macro_tool/README.md

```diff
@@ -1,15 +1,74 @@
-A tool for running macro examples end to end.
+# macro_tool
 
-## Example usage
+A tool for running and benchmarking with macros.
 
-The following will run the json_codable example, assuming you start at the root
-of this repo:
+Specify a list of commands in addition to options:
+
+  apply: runs macros, outputs to disk
+  patch_for_analyzer: fixes macro output for the analyzer
+  patch_for_cfe: fixes macro output for the CFE
+  run: runs the specified script
+  revert: reverts patches to files
+  benchmark_apply: benchmarks applying macros
+  benchmark_analyze: benchmarks analyzing macros
+  bust_caches: modifies files like the benchmarks do
+  watch: loops watching for changes to the specified script and applying
+
+## Examples
+
+All examples are run from the root of this repo.
+
+Benchmarks required that you first create sources to benchmark with:
+
+```bash
+dart tool/benchmark_generator/bin/main.dart large macro 16
```

Comment:
This gives much faster re-compilation of the script itself:
```suggestion
dart run benchmark_generator:main large macro 16
```

Same for the examples below